libraries/chain/controller.cpp

```diff
@@ -1130,9 +1130,19 @@ struct controller_impl {
       });
    }
 
-   bool fork_db_validated_block_exists( const block_id_type& id ) const {
+   bool fork_db_validated_block_exists( const block_id_type& id ) {
```

Comment:
This is not commented as thread safe and is currently only called from the main thread. But it used to be thread-safe, but is no longer for `block_state_legacy`.

Reply:
Why is it not thread safe?

Reply:
`is_valid()` is not thread safe for `block_state_legacy`

Reply:
Shall we make it a `copyable_atomic<bool>` as well?

Reply:
Not of 1.0.x I don't think.

Reply:
Other uses of `is_valid` that are think are not restricted to `block_state`:
[1](https://github.com/AntelopeIO/spring/blob/c4cd685edc3fd9a604ec2d431313a91aeaaa502b/libraries/chain/controller.cpp#L1510)  [2](https://github.com/AntelopeIO/spring/blob/c4cd685edc3fd9a604ec2d431313a91aeaaa502b/libraries/chain/controller.cpp#L1825) [3](https://github.com/AntelopeIO/spring/blob/c4cd685edc3fd9a604ec2d431313a91aeaaa502b/libraries/chain/controller.cpp#L3673) [4](https://github.com/AntelopeIO/spring/blob/c4cd685edc3fd9a604ec2d431313a91aeaaa502b/libraries/chain/controller.cpp#L4402)  [5](https://github.com/AntelopeIO/spring/blob/c4cd685edc3fd9a604ec2d431313a91aeaaa502b/libraries/chain/controller.cpp#L1358)

Reply:
Those are all on the main thread.